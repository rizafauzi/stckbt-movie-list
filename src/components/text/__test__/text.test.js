

import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom'
import Text from '../text.component'
import renderer from 'react-test-renderer'
import { render, cleanup } from '@testing-library/react'

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<Text />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders text correctly", () => {
  const { getByTestId } = render(<Text children="Test Text"/> )
  expect(getByTestId('text')).toHaveTextContent('Test Text')
})


it('it matches snapshot', () => {
  const tree = renderer.create(<Text />).toJSON()
  expect(tree).toMatchSnapshot()
})