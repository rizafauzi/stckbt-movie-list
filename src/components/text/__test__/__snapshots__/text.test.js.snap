// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<h1
  className="sc-bdnylx eGJSLs"
  data-testid="text"
/>
`;
